### PR TITLE
Add the current Mesos host as a Memberlist seed

### PR DESCRIPTION
**Pull request by mihaitodor**
... otherwise, just use 127.0.0.1:7946 as the default Memberlist cluster seed.

Also set the Memberlist advertise address and port when running under Mesos.

### CODE REVIEW

File: docker/s6/services/lazyraster.svc/run [hunk 1]
```diff
@@ -1,5 +1,10 @@
 #!/bin/sh
 
+# Add the current host as a Memberlist cluster seed when running under Mesos
+MESOS_ML_PORT_VAR="MESOS_PORT_${MEMBERLIST_PORT:-7946}"
+MESOS_ML_PORT="${!MESOS_ML_PORT_VAR}"
```
Comment (mihaitodor) [inline docker/s6/services/lazyraster.svc/run]: Looks like this won't work under `sh`. Would it be OK to switch to `bash`?

File: docker/s6/services/lazyraster.svc/run [hunk 2]
```diff
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+# Add the current host as a Memberlist cluster seed when running under Mesos
+MESOS_ML_PORT=$(env | grep "MESOS_PORT_${MEMBERLIST_PORT:-7946}" | cut -d "=" -f 2)
```
Comment (mihaitodor) [inline docker/s6/services/lazyraster.svc/run]: This works, but it's a bit hackish. I can also switch the script to `/bin/bash` and use a dynamic variable instead. WDYT?

File: main.go [hunk 3]
```diff
@@ -17,16 +17,18 @@ import (
...
-	ClusterName  string   `envconfig:"CLUSTER_NAME" default:"default"`
+	BaseDir                 string   `envconfig:"BASE_DIR" default:"."`
+	Port                    string   `envconfig:"PORT" default:"8000"`
+	AwsRegion               string   `envconfig:"AWS_REGION" default:"us-west-1"`
+	S3Bucket                string   `envconfig:"S3_BUCKET" default:"nitro-junk"`
+	ClusterSeeds            []string `envconfig:"CLUSTER_SEEDS" default:"127.0.0.1"`
+	CacheSize               int      `envconfig:"CACHE_SIZE" default:"512"`
+	RedisPort               int      `envconfig:"REDIS_PORT" default:"6379"`
+	ClusterName             string   `envconfig:"CLUSTER_NAME" default:"default"`
+	MemberlistAdvertiseAddr string   `envconfig:"MEMBERLIST_ADVERTISE_ADDRESS"`
```
Comment (relistan) [inline main.go]: Let's not manipulate the names. Just make it `MEMBERLIST_ADVERTISE_ADDR`. It reduces chances of getting it wrong if they match the struct fields exactly.

File: docker/s6/services/lazyraster.svc/run [hunk 4]
```diff
@@ -1,5 +1,20 @@
 #!/bin/sh
 
+if [ -n "${MESOS_HOSTNAME}" ]; then
```
Comment (relistan) [inline docker/s6/services/lazyraster.svc/run]: Can we do all of this in the service instead? Seems kind of janky to have to do it in shell script when the service is designed to be a 12 factor app.

**Comment by mihaitodor** [comment]
@relistan OK, I have made the required changes. Please have another look.

File: main.go [hunk 5]
```diff
@@ -17,16 +19,39 @@ import (
...
-	S3Bucket     string   `envconfig:"S3_BUCKET" default:"nitro-junk"`
-	ClusterSeeds []string `envconfig:"CLUSTER_SEEDS" default:"127.0.0.1"`
-	CacheSize    int      `envconfig:"CACHE_SIZE" default:"512"`
-	RedisPort    int      `envconfig:"REDIS_PORT" default:"6379"`
-	ClusterName  string   `envconfig:"CLUSTER_NAME" default:"default"`
+	BaseDir                 string   `envconfig:"BASE_DIR" default:"."`
+	Port                    string   `envconfig:"PORT" default:"8000"`
+	AwsRegion               string   `envconfig:"AWS_REGION" default:"us-west-1"`
+	S3Bucket                string   `envconfig:"S3_BUCKET" default:"nitro-junk"`
+	ClusterSeeds            []string `envconfig:"CLUSTER_SEEDS"`
```
Comment (mihaitodor) [inline main.go]: Note that I removed the default value from here, since I'm adding it below anyway.

**Comment by mihaitodor** [comment]
@relistan I fixed the stuff we discussed.

**Comment by relistan** [comment]
:shipit:

**Pull request merged**